src/ui/QueryForCommerce/QueryForCommerce.ts

```diff
@@ -0,0 +1,71 @@
+import { Component } from '../Base/Component';
+import { IComponentBindings } from '../Base/ComponentBindings';
+import { ComponentOptions } from '../Base/ComponentOptions';
+import { Assert } from '../../misc/Assert';
+import { QueryEvents, IBuildingQueryEventArgs } from '../../events/QueryEvents';
+import { AnalyticsEvents, IChangeAnalyticsCustomDataEventArgs } from '../../events/AnalyticsEvents';
+import { Initialization } from '../Base/Initialization';
+import { exportGlobally } from '../../GlobalExports';
+
+export interface IQueryForCommerceOptions {
+  listing?: string;
+}
+
+/**
+ * The QueryForCommerce component enables to set properties for listing pages in commerce sites in a convenient way
+ */
+export class QueryForCommerce extends Component {
+  static ID = 'QueryForCommerce';
+
+  static doExport = () => {
+    exportGlobally({
+      QueryForCommerce: QueryForCommerce
+    });
+  };
+
+  /**
+   * The options for the QueryForCommerce.
+   * @componentOptions
+   */
+  static options: IQueryForCommerceOptions = {
+    /**
+     * Setting this property will update the tab value of a query and set the originLevel2 in the analytics event
+     *
+     * This is useful for listing pages in commerce sites to leverage the Coveo ML features.
```

Comment:
This will also need a lot more "meat" around it.

Again @fbeaudoincoveo we can discuss live about the reason for the existence of the listing property. Vincent Bernard/Alex Moreau would also have pertinent information to explain the context around the need for this property and how it gets ultimately leveraged in the query pipelines/ML